### PR TITLE
Add and enforce minimum width

### LINKED ISSUE #4976: Minimize button disappears at minimum size
# Environment

```none
Windows build number: [Microsoft Windows [Version 10.0.18363.535]]
Windows Terminal version (if applicable): 0.10.761.0
```

# Steps to reproduce

Resize the terminal to the minimum size.

# Expected behavior
Minimize button should be visible/A minimum size limit like other apps:
Chrome: image

Credge: image

Firefox: image

Notepads: image

# Actual behavior
Just like most win32 apps, the terminal becomes too small(Also look how near the close button is to the edge. Some grid layout issues I guess):
image

Other programs like explorer, task manager etc show the same behaviour(atleat explorer has a minimum size limit):
image

image

Interestingly, you can see that a minimize button drawn by the system(DWM?) can be seen in the screenshot just outside the windoe. Some other win32 apps like task manager have it too.But from what I understand it should not be there since we are drawing our own title bar (Using Xaml Islands?). Also, if I try to take a screenshot of only the window by using alt+prtsc, the minimize button does not appear.

I think we should have a minimum size limit like the browsers, or if not, atleast the (caption?) buttons should be fully visible at minimum size.

### PR DESCRIPTION
**Pull request by carlos-zamora**
## Summary of the Pull Request
Enforces a minimum width of 460 wide. This allows tabs to always be seen (albeit partially).

NOTE: A minimum height was originally added to allow the About menu to be seen. This minimum height was removed from this PR because we don't foresee a circumstance where your Terminal is in that state, and you can't just resize it to make the content fit properly.

## References
#4990 may be fixed/affected by this?

## PR Checklist
* [X] Closes #4976 
* [X] Will close issue #5418 upon verification

## Validation Steps Performed
Here's some images of what the terminal looks like when it's at its minimum size:

#### 100% scaling: one tab
100% scaling: one tab

#### 100% scaling: two tabs
100% scaling: two tabs

#### 100% scaling: 3 tabs (scrolling enabled)
100% scaling: 3 tabs (scrolling enabled)

#### 200% scaling: one tab
200% scaling: one tab

#### 200% scaling: two tabs
200% scaling: two tabs

#### 200% scaling: 3 tabs (scrolling enabled)
200% scaling: 3 tabs (scrolling enabled)

### CODE REVIEW

File: src/cascadia/WindowsTerminal/IslandWindow.cpp [hunk 1]
```diff
@@ -179,9 +179,24 @@ LRESULT IslandWindow::_OnSizing(const WPARAM wParam, const LPARAM lParam)
     // If this fails, we'll use the default of 96.
     GetDpiForMonitor(hmon, MDT_EFFECTIVE_DPI, &dpix, &dpiy);
 
+    // The minimum client width that you can resize to.
+    // This lets the tabs fit.
+    const auto minClientWidth = 460;
```
Comment (DHowett-MSFT) [inline src/cascadia/WindowsTerminal/IslandWindow.cpp:184]: Can you make this a global static constexpr til::size?

Comment (DHowett-MSFT) [reply src/cascadia/WindowsTerminal/IslandWindow.cpp:184]: then you can use the size scaling operators.

Comment (carlos-zamora) [reply src/cascadia/WindowsTerminal/IslandWindow.cpp:184]: Adding a global static constexpr til::size.

Can't use the scaling operators though :( scaling is a float and you can't have a til::point/size be a float.

**Comment by DHowett-MSFT** [comment]
```
        template<typename TilMath>
        size scale(TilMath, const float scale) const
        {
            struct
            {
                float Width, Height;
            } sz;
            THROW_HR_IF(E_ABORT, !base::CheckMul(scale, _width).AssignIfValid(&sz.Width));
            THROW_HR_IF(E_ABORT, !base::CheckMul(scale, _height).AssignIfValid(&sz.Height));

            return til::size(TilMath(), sz);
        }

```

???????? definitely exists

sz.scale(til::math::rounding, 1.5f)

**Comment by carlos-zamora** [comment]
> ```
>         template<typename TilMath>
>         size scale(TilMath, const float scale) const
>         {
>             struct
>             {
>                 float Width, Height;
>             } sz;
>             THROW_HR_IF(E_ABORT, !base::CheckMul(scale, _width).AssignIfValid(&sz.Width));
>             THROW_HR_IF(E_ABORT, !base::CheckMul(scale, _height).AssignIfValid(&sz.Height));
> 
>             return til::size(TilMath(), sz);
>         }
> ```
> 
> ???????? definitely exists
> 
> sz.scale(til::math::rounding, 1.5f)

Yeah. Michael just gave me a walkthrough on it and being able to cast the output. I've been out of the loop for a while haha.

**Comment by carlos-zamora** [comment]
> I concur with:
> 
> * [x]  Please make sure this applies when `snapToGridOnResize` is false.

Just verified it, works fine.

> * [x]  Maybe we should only enforce a minimum width.

#5418 needs the width and height. Width ensures tabs can always be read. Height ensures about menu fits (see attached features above.

> * [ ]  Does this actually resolve #4990?

I can't actually get a repro on this one. But I can't get it on 0.11 either. @DHowett-MSFT or @zadjii-msft, if either of you can actually get a repro on this, would you mind checking it? Mainly just tagged it because there's a comment that enforcing a minimum size _might_ help.

**Comment by carlos-zamora** [comment]
Here's an idea of how big the minimum size is:
minimum size example

**Comment by DHowett-MSFT** [comment]
This is too tall. people will hate it.

**Comment by DHowett-MSFT** [comment]
Some people legitimately want a one-line-tall terminal. Let's not optimize around an ephemeral dialog.

**Comment by carlos-zamora** [comment]
I hate it too :/ But how else can we get the About menu to fit?

**Comment by zadjii-msft** [comment]
Can we do this for now, and add a minimum size setting in 2.0?





________________________________

**Comment by DHowett-MSFT** [comment]
I'm a hard NAK on the height. Width is fine. Min height = tab strip height. Let people make their about dialogs disappear.

**Comment by carlos-zamora** [comment]
Updated the PR to only enforce the minimum width.

**Comment by DHowett-MSFT** [comment]
Thanks

**Pull request merged**